src/rawstreamreader.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 Veloman Yunkan
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#ifndef ZIM_RAWSTREAMREADER_H
+#define ZIM_RAWSTREAMREADER_H
+
+#include "istreamreader.h"
+#include "reader.h"
+#include "debug.h"
+
+namespace zim
+{
+
+class RawStreamReader : public IStreamReader
+{
+public: // functions
+  explicit RawStreamReader(std::shared_ptr<const zim::Reader> reader)
+    : m_reader(reader),
+      m_readerPos(0)
+  {}
+
+  void readImpl(char* buf, zsize_t nbytes) override
+  {
+    m_reader->read(static_cast<char*>(buf), m_readerPos, zsize_t(nbytes));
```

Comment:
The `static_cast` is redundant

Reply:
Indeed. The original `readImpl` was taking a `void*` and I forget about the cast in the implementation.